src/Permissions.ts

```diff
@@ -0,0 +1,214 @@
+    const amountMembers = this.members.byState.added?.size ?? 0
+    // The signature of the member that created the request is not necessary
+    const neededSignatures = amountMembers - 1
+    if (
+      // Remove operations are okay with having one less
+      request.operation === 'remove' &&
+      // Two members can form a majority, to remove one of two members
+      // unilaterally is impossible
+      amountMembers > 2
+    ) {
+      return neededSignatures - 1
+    }
+    return neededSignatures
+  }
+
+  private addSignature (response: Response): number {
+    const signatures = this.signatures.get(response.id)
+    if (signatures === undefined) {
+      this.signatures.set(response.id, new Set(response.from))
+      return 1
+    }
+    if (signatures.has(response.from)) {
+      throw new Error(`${response.from} tried to sign the same request twice`)
+    }
+    signatures.add(response.from)
+    return signatures.size
+  }
+
+  private finishRequest (request: Request): void {
+    if (request.operation === 'merge') {
+      throw new Error('todo')
+    }
+    if (request.operation === 'add') {
+      this.members.set(request.who, 'added')
+    } else {
+      this.members.set(request.who, 'removed')
+    }
+    this.requests.set(request.id, 'finished')
+    this.openRequests.delete(request.id)
+    this.signatures.delete(request.id)
+    const list = this.openRequestsByMember.get(request.from)
+    if (list === undefined) {
+      throw new Error('This may never occur')
+    }
+    if (list.shift() !== request) {
+      throw new Error('This may also never occur')
+    }
+    const entry = list[0]
+    if (entry === undefined) {
+      this.openRequestsByMember.delete(request.from)
+    } else {
+      this.requests.set(entry.id, 'active')
+    }
+  }
+
+  private handleCancel (response: Response, openRequest?: Request): void {
+    if (openRequest !== undefined) {
+      if (openRequest.from !== response.from) {
+        throw new Error(`Member ${response.from} can not cancel the request ${response.id} by ${openRequest.from}.`)
+      }
+      this.requests.set(response.id, 'cancelled')
+      this.openRequests.delete(response.id)
+      this.openRequestsByMember.delete(response.id)
+    }
+    // TODO: Should we throw an error if the request is not active?
+  }
+
+  private handleDeny (response: Response, openRequest?: Request): void {
+    if (openRequest !== undefined) {
+      if (openRequest.from === response.from) {
+        throw new Error(`Member ${response.from} can not deny their own request ${response.id}. Maybe they meant to cancel?`)
+      }
+      this.requests.set(response.id, 'denied')
+      this.openRequests.delete(response.id)
+      this.openRequestsByMember.delete(response.id)
+    }
+    // TODO: Should we throw an error if the request is not active?
+  }
+
+  private handleRequest (request: Request): void {
+    if (request.operation === 'merge') {
```

Comment:
I think it makes sense to remove merge requests from the code until we have a reason to use them again.

Reply:
:+1: I think these should be removed.